extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueBridgeNupnpDiscovery.java

```diff
@@ -0,0 +1,141 @@
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link HueBridgeNupnpDiscovery} is responsible for discovering new
+ * hue bridges. It uses the 'NUPnP service provided by Philips'.
+ *
+ * @author Awelkiyar Wehabrebi - Initial contribution
+ *
+ */
+public class HueBridgeNupnpDiscovery extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(HueBridgeNupnpDiscovery.class);
+
+    private static final String url = "https://www.meethue.com/api/nupnp";
+
+    private static final int timeout = 5000;
+
+    private static final int BRIDGE_SCAN_INTERVAL = 60;
+
+    private ScheduledFuture<?> bridgeScanningJob;
+
+    public HueBridgeNupnpDiscovery() {
+        super(SUPPORTED_THING_TYPES_UID, 10);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (bridgeScanningJob == null || bridgeScanningJob.isCancelled()) {
+            this.bridgeScanningJob = scheduler.scheduleAtFixedRate(new Runnable() {
+                @Override
+                public void run() {
+                    discoverHueBridges();
+                }
+            }, 0, BRIDGE_SCAN_INTERVAL, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        if (bridgeScanningJob != null || !bridgeScanningJob.isCancelled()) {
+            bridgeScanningJob.cancel(true);
+            bridgeScanningJob = null;
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        discoverHueBridges();
+    }
+
+    /**
+     * Discover available Hue Bridges and then add them in the discovery inbox
+     */
+    private void discoverHueBridges() {
+        // get bridge list
+        List<BridgeJsonParameters> bridgeList = getBridgeList();
+        if (bridgeList != null) {
+            for (BridgeJsonParameters bridge : bridgeList) {
+                if (bridge.getId() != null && bridge.getInternalipaddress() != null) {
+                    // check if bridge is reachable, and if 'id' contains 'fffe' at the middle
+                    if (isHueBridge(bridge.getInternalipaddress()) && bridge.getId().substring(6, 10).equals("fffe")) {
+                        // 'fffe' should be removed from 'id', to match with 'id' in description.xml file
+                        ThingUID uid = new ThingUID(THING_TYPE_BRIDGE,
+                                bridge.getId().substring(0, 6) + bridge.getId().substring(10));
+                        if (uid != null) {
+                            Map<String, Object> properties = new HashMap<>(2);
+                            properties.put(HOST, bridge.getInternalipaddress());
+                            properties.put(SERIAL_NUMBER, bridge.getId().replace("fffe", ""));
+                            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                                    .withLabel("Philips hue (" + bridge.getInternalipaddress() + ")")
```

Comment:
My point is that we must make sure that it is exactly the same label - since the Inbox will "flicker" otherwise in case the one or the other discovery service puts a result in the Inbox for the same UID.